inference-engine/src/preprocessing/cpu_x86_avx2/ie_preprocess_gapi_kernels_avx2.cpp

```diff
@@ -131,6 +131,327 @@ void calcRowArea_32F(float dst[], const float *src[], const Size& inSz,
+                                     shuf_mask2, shuf_mask3, idxs, half_nlanes);
+        } else {  // if any lpi
+            int inLength = inSz.width;
+            int outLength = outSz.width;
+            for (int l = 0; l < lpi; ++l) {
+                short beta0 = beta[l];
+
+                // vertical pass
+                GAPI_DbgAssert(inSz.width >= half_nlanes);
+                verticalPass_anylpi_8U(src0, src1, tmp, beta0, half_nlanes, l, inLength, inLength);
+
+                // horizontal pass
+                GAPI_DbgAssert(outSz.width >= half_nlanes);
+                horizontalPass_anylpi_8U(alpha, mapsx, dst, tmp, l, half_nlanes, outLength);
+            }
+        }  // if lpi == 4
+
+    } else if (!xRatioEq) {
+        GAPI_DbgAssert(yRatioEq);
+
+        if (4 == lpi) {
+            // vertical pass
+            GAPI_DbgAssert(inSz.width >= nlanes);
+            for (int w = 0; w < inSz.width; ) {
+                for (; w <= inSz.width - nlanes; w += nlanes) {
+                    v_uint8 s0, s1, s2, s3;
+                    s0 = vx_load(&src0[0][w]);
+                    s1 = vx_load(&src0[1][w]);
+                    s2 = vx_load(&src0[2][w]);
+                    s3 = vx_load(&src0[3][w]);
+                    v_store_interleave(&tmp[4 * w], s0, s1, s2, s3);
+                }
+
+                if (w < inSz.width) {
+                    w = inSz.width - nlanes;
+                }
+            }
+
+            // horizontal pass
+            v_uint8 val_0, val_1, val_2, val_3;
+
+            v_uint8 shuf_mask2 = v_setr_s8(0, 4, 8, 12, 2, 6, 10, 14,
+                                           1, 5, 9, 13, 3, 7, 11, 15,
+                                           0, 4, 8, 12, 2, 6, 10, 14,
+                                           1, 5, 9, 13, 3, 7, 11, 15);
+            v_uint8 shuf_mask3 = v_setr_s8(0, 1, 8,  9,  2, 3, 10, 11,
+                                           4, 5, 12, 13, 6, 7, 14, 15,
+                                           0, 1, 8,  9,  2, 3, 10, 11,
+                                           4, 5, 12, 13, 6, 7, 14, 15);
+            v_uint32 idxs = v_setr_s32(0, 2, 4, 6, 1, 3, 5, 7);
+
+            GAPI_DbgAssert(outSz.width >= half_nlanes);
+            horizontalPass_lpi4_8UC1(clone, mapsx, tmp, dst, outSz.width,
+                                     val_0, val_1, val_2, val_3,
+                                     shuf_mask2, shuf_mask3, idxs, half_nlanes);
+        } else {  // any LPI
+            for (int l = 0; l < lpi; ++l) {
+                const uchar *src = src0[l];
+
+                // horizontal pass
+                GAPI_DbgAssert(outSz.width >= half_nlanes);
+                horizontalPass_anylpi_8U(alpha, mapsx, dst, src, l, half_nlanes, outSz.width);
+            }
+        }
+
+    } else if (!yRatioEq) {
+        GAPI_DbgAssert(xRatioEq);
+        int inLength = inSz.width;
+        int outLength = outSz.width;
```

Comment:
Not sure if we need this temporary variables here, inSz.width and outSz.width can be used on their own